src/Processor/FillMissingProcessor.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the BibTex Parser.
+ *
+ * (c) Renan de Lima Barbosa <renandelima@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace RenanBr\BibTexParser\Processor;
+
+class FillMissingProcessor
+{
+    use TagSearchTrait;
+
+    protected $missingFields;
+
+    public function __construct($missingFields)
+    {
+        $this->missingFields = $missingFields;
+    }
+
+    public function __invoke(array $entry)
+    {
+        foreach ($this->missingFields as $tag => $value) {
+            if(!$this->tagSearch($tag, $entry)) {
```

Comment:
`tagSearch()` expects as 2nd argument a list of tag names, not the entry itself :wink:

I think storing `array_keys()` result to a variable before the loop solves the case